uint16 gpioId
uint16 pulseLengthInUs
---
bool hasSucceeded